rust-book-old/src/toolchain/katana/interact.md

```diff
@@ -0,0 +1,321 @@
+### install starkli
+
+```console
+curl https://get.starkli.sh | sh
+## open a new terminal
+starkliup
+```
+
+You can check your installation by running `starkli --version`,then you will get the starkli version.
+
+### install katana
+
+```console
+git clone https://github.com/dojoengine/dojo
+cd dojo
+cargo install --path ./crates/katana --locked --force
+```
+
+You can check your installation by running `katana --version`,then you will get the katana version.
+
+### Deploy and interact with contracts
+
+#### run katana node
+
+```console
+katana --accounts 3 --seed 0 --gas-price 250
+```
+
+After starting the node, the accounts you need will be automatically generated and deployed. We need to import the pre-deployed accounts into starkli for use.
+
+#### Configure starkli account
+
+1. Configure the signer, execute the following command, and then enter the private key. This private key is the private key of the account generated by kanata.
+
+```console
+starkli signer keystore from-key ~/.starkli-wallets/deployer/account0_keystore.json
+```
+
+2. Configure account description and create a new json file
+
+```console
+touch ~/.starkli-wallets/deployer/account0_account.json
+```
+
+3. Copy the following content into the json file
+
+```console
+{
+  "version": 1,
+  "variant": {
+        "type": "open_zeppelin",
+        "version": 1,
+        "public_key": "<SMART_WALLET_PUBLIC_KEY>"
+  },
+    "deployment": {
+        "status": "deployed",
+        "class_hash": "<SMART_WALLET_CLASS_HASH>",
+        "address": "<SMART_WALLET_ADDRESS>"
+  }
+}
+```
+
+Change `<>` and its contents to those provided by katana, including `public_key`, `class_hash`, and `address`.Among them, `class_hash` needs to be obtained with the following command. Please replace `<SMART_WALLET_ADDRESS>` with address.
+
+```console
+starkli class-hash-at <SMART_WALLET_ADDRESS> --rpc http://0.0.0.0:5050
+```
+
+In order to facilitate subsequent contract deployment and invocation, three accounts are deployed directly here. The process is the same.we will get 6 json files.
+
+#### Contract Deployment
+
+1. Create a Vote project
+
+```console
+## create a new project
+scarb new vote
+## Add contract dependencies to scarb.toml
+[dependencies]
+starknet = "2.1.0"
```

Comment:
Currently, it's recommended to use at most starknet `2.3.1` with scarb `2.3.1` to be compatible with Katana.